deploy.py

```diff
@@ -166,38 +292,104 @@ def update_rustc_if_needed(self):
       # Need to update
       self.checked_command_output(
           ["rustup", "install", target_toolchain_fullstring])
-      self.checked_command_output(
-          ["rustup", "target", "add", "thumbv7em-none-eabi"])
+    targets = set([x.arch for _, x in SUPPORTED_BOARDS.items()])
+    for arch in targets:
+      self.checked_command_output(["rustup", "target", "add", arch])
     info("Rust toolchain up-to-date")
 
-  def build_and_install_tockos(self):
+  def search_binary(self, name, start_directory="."):
+    for root, _, files in os.walk(start_directory):
+      for fname in files:
+        if fname == name:
+          return os.path.join(root, fname)
+    return None
+
+  def build_tockos(self):
```

Comment:
I don't think we should manually build things instead of relying on Tock's Makefiles. That's too much to maintain for us and there are pending changes that affect it (tock/tock#1660, tock/tock#1668, tock/tock#1669).

We should rather have some patches to `third_party/tock` that add the relevant boards.

Reply:
But then it means we need to have a Makefile for every out-of-tree board that we want to support.

Reply:
Will these Makefiles be any different from one another? The `./setup.sh` script should take care of that.

Reply:
Moved back to using make.

Reply:
Done